spatialmath/box.go

```diff
@@ -307,11 +307,11 @@ func boxVsBoxCollision(a, b *box) bool {
 // references:  https://comp.graphics.algorithms.narkive.com/jRAgjIUh/obb-obb-distance-calculation
 //
 //	https://dyn4j.org/2010/01/sat/#sat-nointer
-func boxVsBoxDistance(a, b *box) float64 {
+func boxVsBoxDistance(a, b *box, collisionBufferMM float64) float64 {
 	centerDist := b.pose.Point().Sub(a.pose.Point())
 
 	// check if there is a distance between bounding spheres to potentially exit early
-	if boundingSphereDist := centerDist.Norm() - a.boundingSphereR - b.boundingSphereR; boundingSphereDist > CollisionBuffer {
```

Comment:
Good call out Nick that this is weird to be changing.  I agree this function shouldnt really have to consider this.  Anything upstream of this I imagine would take the output of the distance function and use the collision buffer in its logic.  @biotinker thoughts?

Reply:
Per my other comment; the reasoning behind the use of `CollisionBuffer` here was to provide deterministic floating point behavior across different architectures for precisely-touching geometries. We probably want to restore the const for use here for that same reason.

Also, something I think we should be emphasizing here, is that a positive number returned from `DistanceFrom` is saying "these geometries are *at least* this far apart", not that they are exactly this far apart.

Reply:
I am taking over development for this ticket and have reverted back to the constant we had before.  Do you think I should add a comment emphasizing your second point?